### PR TITLE
feat: Tag support cssVar

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- #45618

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: Tag support cssVar |
| 🇨🇳 Chinese | feat: Tag 组件支持 cssVar |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 62e9c1f</samp>

This pull request refactors the Tag component to use CSS variables for styling, and adds a new hook `useCSSVar` to register and apply them. It also improves the type safety and consistency of the style modules and functions for the Tag component.

### <samp>🤖 Generated by Copilot at 62e9c1f</samp>

*  Add `useCSSVar` hook to register and update CSS variables for Tag component ([link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-bcceb016a2b5f77c9c373d92bb6e81d64f26323154b13adfc6b74243e78b4913R14), [link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-c2f5a27137d106bd7ff568f0ad5ce8f374d9d7ffa0f351ff74f411f6951a06feR1-R4))
*  Use `useCSSVar` hook to wrap Tag component with CSS variables and remove `wrapSSR` function ([link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-bcceb016a2b5f77c9c373d92bb6e81d64f26323154b13adfc6b74243e78b4913L82-R85), [link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-bcceb016a2b5f77c9c373d92bb6e81d64f26323154b13adfc6b74243e78b4913L149-R151))
*  Convert numeric style values to CSS unit strings using `unit` function in `genBaseStyle` function ([link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-cb2262b77ae285e7da02504971f378bec871797a530abbf2a34f14fdb3d92f85L45-R51), [link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-cb2262b77ae285e7da02504971f378bec871797a530abbf2a34f14fdb3d92f85L68-R69), [link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-cb2262b77ae285e7da02504971f378bec871797a530abbf2a34f14fdb3d92f85L119-R118))
*  Rename `prepareCommonToken` function to `prepareComponentToken` and use it to prepare component token for Tag component in `useStyle`, `usePresetStyle`, and `useStatusStyle` hooks ([link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-cb2262b77ae285e7da02504971f378bec871797a530abbf2a34f14fdb3d92f85L146-R145), [link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-cb2262b77ae285e7da02504971f378bec871797a530abbf2a34f14fdb3d92f85L153-R158), [link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-79f09cde60f663f18ad398ac97f9dffdd51e3ff6c851cfc508e5cfdb51cc11d9L2-R2), [link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-79f09cde60f663f18ad398ac97f9dffdd51e3ff6c851cfc508e5cfdb51cc11d9L25-R31), [link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-b41be4b94be27b2254289f98a0514add547f0e2c93ec2ebb3236111832a702c4L4-R4), [link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-b41be4b94be27b2254289f98a0514add547f0e2c93ec2ebb3236111832a702c4L30-R33), [link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-b41be4b94be27b2254289f98a0514add547f0e2c93ec2ebb3236111832a702c4L42-R41))
*  Add generic type parameter `'Tag'` to `genComponentStyleHook` and `genSubStyleComponent` function calls to specify component name as a type ([link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-cb2262b77ae285e7da02504971f378bec871797a530abbf2a34f14fdb3d92f85L153-R158), [link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-79f09cde60f663f18ad398ac97f9dffdd51e3ff6c851cfc508e5cfdb51cc11d9L25-R31), [link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-b41be4b94be27b2254289f98a0514add547f0e2c93ec2ebb3236111832a702c4L30-R33), [link](https://github.com/ant-design/ant-design/pull/45823/files?diff=unified&w=0#diff-b41be4b94be27b2254289f98a0514add547f0e2c93ec2ebb3236111832a702c4L42-R41))

### CODE REVIEW

File: components/tag/style/index.ts [hunk 1]
```diff
@@ -42,14 +41,14 @@ const genBaseStyle = (token: TagToken): CSSInterpolation => {
       ...resetComponent(token),
       display: 'inline-block',
       height: 'auto',
-      marginInlineEnd: token.marginXS,
-      paddingInline,
-      fontSize: token.tagFontSize,
+      marginInlineEnd: unit(token.marginXS),
+      paddingInline: unit(paddingInline),
+      fontSize: unit(token.tagFontSize),
```
Comment (MadCcc) [inline components/tag/style/index.ts:46]:
```suggestion
      marginInlineEnd: token.marginXS,
      paddingInline,
      fontSize: token.tagFontSize,
```

File: components/tag/style/index.ts [hunk 2]
```diff
@@ -42,14 +41,14 @@ const genBaseStyle = (token: TagToken): CSSInterpolation => {
...
       lineHeight: token.tagLineHeight,
       whiteSpace: 'nowrap',
       background: token.defaultBg,
-      border: `${token.lineWidth}px ${token.lineType} ${token.colorBorder}`,
-      borderRadius: token.borderRadiusSM,
+      border: `${unit(token.lineWidth)} ${token.lineType} ${token.colorBorder}`,
+      borderRadius: unit(token.borderRadiusSM),
```
Comment (MadCcc) [inline components/tag/style/index.ts:51]:
```suggestion
      borderRadius: token.borderRadiusSM,
```

File: components/tag/style/index.ts [hunk 3]
```diff
@@ -65,9 +64,9 @@ const genBaseStyle = (token: TagToken): CSSInterpolation => {
       },
 
       [`${componentCls}-close-icon`]: {
-        marginInlineStart: iconMarginInline,
+        marginInlineStart: unit(iconMarginInline),
```
Comment (MadCcc) [inline components/tag/style/index.ts:67]:
```suggestion
        marginInlineStart: iconMarginInline,
```

File: components/tag/style/index.ts [hunk 4]
```diff
@@ -65,9 +64,9 @@ const genBaseStyle = (token: TagToken): CSSInterpolation => {
...
+        fontSize: unit(token.tagIconSize),
```
Comment (MadCcc) [inline components/tag/style/index.ts:68]:
```suggestion
        fontSize: token.tagIconSize,
```

File: components/tag/style/index.ts [hunk 5]
```diff
@@ -116,7 +115,7 @@ const genBaseStyle = (token: TagToken): CSSInterpolation => {
 
       // To ensure that a space will be placed between character and `Icon`.
       [`> ${token.iconCls} + span, > span + ${token.iconCls}`]: {
-        marginInlineStart: paddingInline,
+        marginInlineStart: unit(paddingInline),
```
Comment (MadCcc) [inline components/tag/style/index.ts:118]:
```suggestion
        marginInlineStart: paddingInline,
```

File: components/tag/style/index.ts [hunk 6]
```diff
@@ -143,19 +142,18 @@ export const prepareToken: (token: Parameters<GenStyleFn<'Tag'>>[0]) => TagToken
   return tagToken;
```
Comment (MadCcc) [inline components/tag/style/index.ts:142]: prepareToken 要改

Comment (li-jia-nan) [reply components/tag/style/index.ts:139]: done

**Review by MadCcc — APPROVED** [review]

**Pull request merged**